docs/container.md

```diff
@@ -0,0 +1,69 @@
+<!--
+SPDX-FileCopyrightText: Red Hat, Inc.
+SPDX-License-Identifier: GPL-2.0-or-later
+-->
+
+# Containerized imageio server
+
+## Overview
+
+ovirt-imageio provides an image for running the ovirt-imageio-daemon
+in a container.
+
+Using a container, users can run an image transfer pod for every
+transfer, exposing the pod for remote access, and connecting the
+relevant storage the pod. When the transfer is done, the user will
+unexpose the pod and terminate it.
+
+In this mode the imageio server will serve one ticket, injected into the
+pod when starting the pod, instead of adding a ticket using the control
+service. The control service is actually disabled.
+
+To create the container run:
+
+    make container
+
+This creates localhost/ovirt-imageio container image using podman
+configured for [rootless environment](https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md).
+
+
+To run the container we need to load a ticket and create the image in the
+mounted images folder. Note that the path for the file in the ticket needs
+to match the path where we mounted the images folder in the container.
+
+Alternatively, we can mount the nbd socket path in the container
+and request the image with a nbd ticket.
+
+In order to specify the path to the ticket that we want to load,
+you can use an environment variable, namely `TICKET_PATH`, that will
+allow you to define the full path for the volume mount point
+and the ticket name.
+
+## Example session
+
+```
+    $ mkdir /var/tmp/images
+    $ qemu-img create -f raw /var/tmp/images/disk.raw 6g
+    $ podman run \
+        --interactive \
+        --tty \
+        --rm \
+        --publish 8080:80 \
+        --env TICKET_PATH=/ticket/file.json \
+        --volume ./examples:/ticket:Z \
+        --volume /var/tmp/images:/var/tmp:Z \
+        localhost/ovirt-imageio:latest
+    2022-10-08 14:41:14,961 INFO    (MainThread) [server] Starting (hostname=51365cc84a8b pid=1, version=2.4.7)
+    2022-10-08 14:41:14,966 INFO    (MainThread) [services] remote.service listening on ('::', 80)
+    2022-10-08 14:41:14,966 INFO    (MainThread) [server] Initial ticket: /ticket/file.json
+    2022-10-08 14:41:14,967 INFO    (MainThread) [server] Ready for requests
+```
+
+In another shell, download the image:
+
+```
+    $ curl http://localhost:8080/images/file -o /data/tmp/download.raw
+      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100 6144M  100 6144M    0     0  1342M      0  0:00:04  0:00:04 --:--:-- 1347M
+```
```

Comment:
Pointing to the imageio API doc can be useful here. curl works but is very inefficient.
In this case it is downloading 6g of zeroes.

Sadly we don't not support http (only https) in the http module, so we cannot use 
the imageio client library to download this image. Maybe we need to add http support,
it should be pretty trivial (use `HTTPConnection` instead of `HTTPSConnection`) here:
https://github.com/oVirt/ovirt-imageio/blob/57cdb2e34c27a00d5b2c2ece94b6406211d82564/ovirt_imageio/_internal/backends/http.py#L380

and skipping `_create_ssl_context` when using `http` scheme.

Reply:
Sounds good, I'll explore this option in another PR.

Reply:
The http PR should not take much time to complete, so we can wait for a while with the document, and show example download with examples/imageio-client.

Reply:
I tested this with imageio client, and it is actually slower than curl since we don't
support extents for file backend:

    $ examples/imageio-client download -f raw http://localhost:8080/images/file \
        /data/tmp/download.raw
    [ 100% ] 6.00 GiB, 5.73 s, 1.05 GiB/s                                          

    $ curl http://localhost:8080/images/file -o /data/tmp/download.raw
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
    100 6144M  100 6144M    0     0  1269M      0  0:00:04  0:00:04 --:--:-- 1263M

But since imageio client sparsify the downloaded image, we get a sparse file:

    $ qemu-img info /data/tmp/download.raw 
    image: /data/tmp/download.raw
    file format: raw
    virtual size: 6 GiB (6442450944 bytes)
    disk size: 0 B

while curl creates fully allocated file:

    $ qemu-img info /data/tmp/download.raw 
    image: /data/tmp/download.raw
    file format: raw
    virtual size: 6 GiB (6442450944 bytes)
    disk size: 6 GiB

When we enable the nbd backend the results will be much better.